src/lib/chips/chip.ts

```diff
@@ -1,17 +1,154 @@
-import { Component, ElementRef, Renderer } from '@angular/core';
+import {
+  // Classes
+  AfterViewInit,
+  Component,
+  ElementRef,
+  EventEmitter,
+  OnDestroy,
+  Renderer,
 
-@Component({
-  selector: 'md-chip, [md-chip]',
+  // Functions
+  forwardRef
+} from '@angular/core';
+
+import {MdFocusable} from '../core/a11y/list-key-manager';
+
+/*
+ * A reusable component config. Utilized by the MdBasicChip
+ * as well as the child class MdChip.
+ */
+export const MD_BASIC_CHIP_COMPONENT_CONFIG: Component = {
+  selector: 'md-basic-chip, [md-basic-chip]',
   template: `<ng-content></ng-content>`,
   host: {
     // Properties
-    'class': 'md-chip',
     'tabindex': '-1',
-    'role': 'option'
-  }
-})
-export class MdChip {
+    'role': 'option',
+
+    // Attributes
```

Comment:
Are we following the official ng2 style guide? If so, I'll just update to match and remove the comments.